src/main/java/seedu/address/logic/commands/EditTaskCommand.java

```diff
@@ -0,0 +1,128 @@
+            + "[" + PREFIX_TASK_DESCRIPTION + "TASK]...\n"
+            + "Example: " + COMMAND_WORD + " 1 " + " 2 "
+            + PREFIX_TASK_DESCRIPTION + "change bandage";
+
+    public static final String MESSAGE_EDIT_TASK_SUCCESS = "Edited Task: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "Task to edit must be provided.";
+
+    private final Index patientIndex;
+    private final Index taskIndex;
+    private final Task updatedTask;
+
+    /**
+     * @param patientIndex of the patient in the filtered patient list to edit.
+     * @param taskIndex of the task to be edited.
+     * @param updatedTask details to edit for the task.
+     */
+    public EditTaskCommand(Index patientIndex, Index taskIndex, Task updatedTask) {
+        requireNonNull(patientIndex);
+        requireNonNull(taskIndex);
+        requireNonNull(updatedTask);
+
+        this.patientIndex = patientIndex;
+        this.taskIndex = taskIndex;
+        this.updatedTask = updatedTask;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+
+        if (patientIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person patientToEdit = lastShownList.get(patientIndex.getZeroBased());
+        Person editedPatient = createEditedPatient(patientToEdit, taskIndex, updatedTask);
+
+        model.setPerson(patientToEdit, editedPatient);
+        model.updateFilteredPersonList(Model.PREDICATE_SHOW_ALL_PERSONS);
+        return new CommandResult(String.format(MESSAGE_EDIT_TASK_SUCCESS, editedPatient));
+    }
+
+    /**
+     * creates an edited person with the updated Task.
+     *
+     * @param patientToEdit the patient associated with the task to edit.
+     * @param indexOfTask the index of the task to be edited.
+     * @param updatedTask the new task details to be edited with.
+     * @return a person with the updated task details.
+     */
+    private static Person createEditedPatient(Person patientToEdit, Index indexOfTask, Task updatedTask) {
+        assert patientToEdit != null;
+
+        Name name = patientToEdit.getName();
+        Phone phone = patientToEdit.getPhone();
+        Email email = patientToEdit.getEmail();
+        Address address = patientToEdit.getAddress();
+        Set<Tag> tags = patientToEdit.getTags();
+        ArrayList<Task> tasks = patientToEdit.getTasks().getTasks();
+
+        ArrayList<Task> updatedTasks = new ArrayList<>();
+
+        for (int i = 0; i < tasks.size(); i++) {
+            if (i == indexOfTask.getZeroBased()) {
+                updatedTasks.add(updatedTask);
+            } else {
+                updatedTasks.add(tasks.get(i));
+            }
+        }
+
+        TaskList updatedTaskList = new TaskList(updatedTasks);
+
```

Comment:
Would it be better if an `edit` method is implemented in the `TaskList` class so the existing `TaskList` is modified, rather than replacing it with an entirely new `TaskList`? For example, a possible implementation of `TaskList#edit` is to remove the specified `Task` and insert the updated `Task` at the same index. `deleteTask` has a similar implementation i.e. editing the `TaskList` directly instead of replacing the entire `TaskList`. The entire `Person` class is already replaced whenever there are changes to a particular `Task`. If we were to say, implement some sort of undo/redo command (as discussed on Thursday), I think previous states of the `TaskList` can still be retrieved via the `Person` object in some way.

Reply:
tl;dr: we may not need to replace the entire `TaskList` since it is implicitly replaced every time a new `Person` object is created

Reply:
I think it would also be better if the logic of updating the task list is moved to the `TaskList` class and not by the `edit` command. So ideally all the `edit` command does is to call `TaskList#update` with the updated `Task`.